### PR TITLE
Create the DB package for use in other services

### LINKED ISSUE #3: Migrate DBUpdaters across JobController and the API to a new package that is depended on by both.

### PR DESCRIPTION
**Pull request by Pasarus**
Closes #3 

## Description
Create a package for import by other python services

### CODE REVIEW

File: db/data_models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    """
+    The Instrument Table's declarative declaration
+    """
+
+    __tablename__ = "instruments"
+    instrument_name: Mapped[str] = mapped_column(unique=True)
+    latest_run: Mapped[str] = mapped_column()
+    specification: Mapped[str] = mapped_column()
+
+    def __eq__(self, other: Self) -> bool:
```
Comment (keiranjprice101) [inline db/data_models/__init__.py]: `other: object`

File: db/data_models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    raw_frames: Mapped[int] = mapped_column(Integer())
+    owner_id: Mapped[int] = mapped_column(ForeignKey("job_owners.id"))
+    owner: Mapped[JobOwner | None] = relationship("JobOwner", lazy="joined")
+    jobs: Mapped[list[Job]] = relationship(
+        secondary=run_job_junction_table,
+        back_populates="runs",
+        lazy="subquery",
+    )
+
+    def __eq__(self, other: Self) -> bool:
```
Comment (keiranjprice101) [inline db/data_models/__init__.py]: `other: object`

File: db/utils/db_updater.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        :param stacktrace: The stacktrace in the event that the db entry failed
+        :return:
+        """
+        with self.session_maker_func() as session:
+            job = session.query(Job).filter_by(id=db_job_id).one()
+            job.state = state
+            job.outputs = str(output_files)
+            job.status_message = status_message
+            job.stacktrace = stacktrace
+            job.start = start  # type: ignore
```
Comment (keiranjprice101) [inline db/utils/db_updater.py:189]: Do these 2 type ignores do anything?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+[project]
+name = "db"
+readme = "README.md"
+version = "0.0.1"
+requires-python = ">= 3.11"
+dependencies = [
+    "psycopg2-binary==2.9.9",
```
Comment (Pasarus) [inline pyproject.toml]:
```suggestion
    "psycopg2==2.9.9",
```

**Review by keiranjprice101 — CHANGES_REQUESTED** [review]
Overall it looks nice and simple and unified 👍 

Couple points about the override type narrowing from object to Self and a couple of ignores that don't look necessary.

Do you plan to adding a workflow for ruff and mypy?

Also the `__eq__` method in the `Base` does that fail in the classes in subclasses where `__eq__` has overrides. It would be nice to have some tests for this anyway.

File: db/utils/db_updater.py (hunk 3 above)
Comment (Pasarus) [reply db/utils/db_updater.py:189]: Apparently necessary as MyPy would otherwise raise, copied from previous implementation.

**Review by keiranjprice101 — APPROVED** [review]
Overall it looks nice and simple and unified 👍 

Couple points about the override type narrowing from object to Self and a couple of ignores that don't look necessary.

Do you plan to adding a workflow for ruff and mypy?

Also the `__eq__` method in the `Base` does that fail in the classes in subclasses where `__eq__` has overrides. It would be nice to have some tests for this anyway.

**Pull request merged**